kore/test/Test/Kore/Step/Simplification/Condition.hs

```diff
@@ -110,6 +120,18 @@ test_simplify_local_functions =
             actual <- simplify condition
             assertBool "Expected \\bottom" $ isBottom actual
 
+    testDefined name eitherFunc eitherC1 eitherC2 =
+        testCase name $ do
+            let equals1 =
+                    mkLocalDefn (applyDefined1 eitherFunc) eitherC1
+                        & Condition.fromPredicate
+                equals2 =
+                    mkLocalDefn (applyDefined2 eitherFunc) eitherC2
+                        & Condition.fromPredicate
+                condition = definedFg <> equals1 <> definedFg <> equals2
```

Comment:
Using `definedFg` here actually requires that `eitherFunc` is `Right fg` or `Left fg`, so I would rather we not allow passing in any value there.

Reply:
Do you mean we should remove `definedFg` from `condition`?